### PR TITLE
Implement keep blocks explosion types

### PR DESCRIPTION
**Pull request by DrexHD**
Checks the block interaction type of the explosion packet and removes the toBlow list if the explosion doesn't destroy blocks.
_(Not sure if there is any better way to peek at the block interaction type)_

### CODE REVIEW

File: common/src/main/java/com/viaversion/viabackwards/protocol/protocol1_20_2to1_20_3/rewriter/BlockItemPacketRewriter1_20_3.java [hunk 1]
```diff
@@ -114,18 +114,40 @@ public void handleCraftingShaped(final PacketWrapper wrapper) throws Exception {
...
+                    wrapper.read(Type.BYTE), // Relative Y
+                    wrapper.read(Type.BYTE) // Relative Z
+                };
+            }
+
+            float knockbackX = wrapper.read(Type.FLOAT); // Knockback X
+            float knockbackY = wrapper.read(Type.FLOAT); // Knockback Y
+            float knockbackZ = wrapper.read(Type.FLOAT); // Knockback Z
+
+            Integer blockInteraction = wrapper.read(Type.VAR_INT); // Block interaction type
```
Comment (kennytv) [inline common/src/main/java/com/viaversion/viabackwards/protocol/protocol1_20_2to1_20_3/rewriter/BlockItemPacketRewriter1_20_3.java:132]: make this the primitive int type

File: common/src/main/java/com/viaversion/viabackwards/protocol/protocol1_20_2to1_20_3/rewriter/BlockItemPacketRewriter1_20_3.java [hunk 2]
```diff
@@ -114,18 +114,40 @@ public void handleCraftingShaped(final PacketWrapper wrapper) throws Exception {
...
+            // 0 = keep, 1 = destroy, 2 = destroy_with_decay, 3 = trigger_block
```
Comment (kennytv) [inline common/src/main/java/com/viaversion/viabackwards/protocol/protocol1_20_2to1_20_3/rewriter/BlockItemPacketRewriter1_20_3.java:133]: What's the trigger one used for exactly? Presumably that one also wouldn't destroy blocks, though I can't check at the minute

Comment (DrexHD) [reply common/src/main/java/com/viaversion/viabackwards/protocol/protocol1_20_2to1_20_3/rewriter/BlockItemPacketRewriter1_20_3.java:133]: The trigger type is used by the breeze to toggle blocks like buttons, doors, levers... It doesn't destroy any blocks.
I have done some basic testing with this PR and everything worked as expected!

**Review by kennytv — APPROVED** [review]
Thanks!

**Pull request merged**